ert_shared/tracker/utils.py

```diff
@@ -0,0 +1,68 @@
+import math
+
+
+def calculate_progress(
+    phase, phase_count, finished, queue_running, queue_size, phase_has_run, done_count
+):
+    if finished:
+        return 1.0
+    if not queue_running and phase_has_run:
+        # queue is not running, but it has run for this phase, so it's done
+        return (phase + 1.0) / phase_count
+    else:
+        phase_progress = float(done_count) / queue_size
+        return (phase + phase_progress) / phase_count
+
+
+def format_running_time(runtime):
+    """ @rtype: str """
+    days = 0
+    hours = 0
+    minutes = 0
+    seconds = math.trunc(runtime)
+
+    if seconds >= 60:
+        minutes, seconds = divmod(seconds, 60)
+
+    if minutes >= 60:
+        hours, minutes = divmod(minutes, 60)
+
+    if hours >= 24:
+        days, hours = divmod(hours, 24)
+
+    if days > 0:
+        layout = "Running time: {d} days {h} hours {m} minutes {s} seconds"
+
+    elif hours > 0:
+        layout = "Running time: {h} hours {m} minutes {s} seconds"
+
+    elif minutes > 0:
+        layout = "Running time: {m} minutes {s} seconds"
+
+    else:
+        layout = "Running time: {s} seconds"
+
+    return layout.format(d=days, h=hours, m=minutes, s=seconds)
+
+
+def scale_intervals(reals):
+    scaled_gen = _scale(reals, min_time=1, max_time=5)
+    scaled_det = _scale(reals, min_time=1, max_time=15)
+    return math.trunc(scaled_gen), math.trunc(scaled_det)
+
+
+def _scale(nr_realisations, min_time=5, max_time=15, min_real=1, max_real=500):
+    nr_realisations = min(max_real, nr_realisations)
+    nr_realisations = max(min_real, nr_realisations)
```

Comment:
does not quite get what these to assignments does... ?

Reply:
@oyvindeide wanna chime in? =)

Reply:
ok, got the explanation which gave sense ... just didn't catch it initially when reading the code.

Reply:
Late to the party (and just adding this in case it improves readability), but is the equivalent of `np.clip(number, min_real, max_real)`, or `max(min_real, min(number, max_real))`, the point is to have a hard upper and lower boundary so that we normalize to 0-1 later.

Reply:
yes, I understand - and it makes sense... just me reading through a bit fast and didn't really got the sense of what's going on. One could probably make this obvious by having 
```
func clamp(nr, realisations, max_real, min_real): 
    nr_realisations = min(max_real, nr_realisations)
    nr_realisations = max(min_real, nr_realisations)
    return nr_realisations
```
but - guess that is smak & behag